src/hoc/withTheme.js

```diff
@@ -51,7 +51,7 @@ const wrapWithTheme = (Component: ReactClass<any>) => {
         )
       } else if (styledContext === undefined && themeProp !== undefined) {
         this.setState({ theme: themeProp })
```

Comment:
Since `themeProp` will never default to `this.props.theme` i.e. `defaultProps.theme` actually (see `determineTheme`) this won't actually apply a theme provided by `defaultProps`.

Reply:
themeProp will be this.props.theme if the prop theme is used because `props.theme === defaultProps.theme` (from determineTheme) will make isDefaultTheme false

Reply:
yep, but in case the theme prop is not being passed and `defaultProps.theme` exists this will cause us to not use it and not log a warning.

Instead it'd be great to just use it, as described in the original issue.

I don't think it'd make sense to hide the warning and then only use `this.props.theme` when it's not defaulting

Reply:
Considering the scenario where theme prop is not being passed and `defaultProps.theme` exist, defaultProps will be used by the component. The test `should not break without a ThemeProvier if it has a defaultTheme` validate this with the first expect.
I'm I completely wrong about this?

Reply:
I'm currently seeing this warning in one of my packages, which is a HOC.
https://www.npmjs.com/package/responsive-props

The HOC is using `withTheme`. I tried suppressing the warning by setting the `defaultProp.theme` of the component wrapping the component to an empty object without any luck.

The warning doesn't appear when wrapped with a `ThemeProvider`.